src/curl.cpp

```diff
@@ -346,7 +346,7 @@ long             S3fsCurl::connect_timeout     = 300;  // default
 time_t           S3fsCurl::readwrite_timeout   = 60;   // default
 int              S3fsCurl::retries             = 3;    // default
 bool             S3fsCurl::is_public_bucket    = false;
-string           S3fsCurl::default_acl         = "private";
+string           S3fsCurl::default_acl         = "";
```

Comment:
I tested a deny rule which fails in master and succeeds with this PR as @sqlbot suggested:

```json
{
  "Version":"2012-10-17",
  "Statement":[
    {
      "Sid":"s3fs-567",
      "Effect":"Deny",
      "Principal": "*",
      "Action":["s3:PutObject"],
      "Resource":["arn:aws:s3:::s3fs-567/*"],
      "Condition":{"StringEquals":{"s3:x-amz-acl":["private"]}}
    }
  ]
}
```

I believe that this is the correct behavior and s3fs deviates from other S3 libraries like Apache jclouds which do not send a canned ACL unless specified.  s3fs should support partially-compatible S3 implementations without additional configuration.  If users want the explicit ACL they can configure it via `-o default_acl=private`.  If you both feel strongly about this I will change the value.